sql/pgwire/encoding.go

```diff
@@ -78,7 +59,11 @@ func (b *readBuffer) readUntypedMsg(rd io.Reader) error {
 			size, maxMessageSize)
 	}
 
-	b.reset(size)
+	if len(b.msgBuf) < size {
```

Comment:
not really; If you look at all the methods on `readBuffer`, they advance the slice each time something is read - without msgBuf it would be impossible to recover the front of the buffer.